src/main/java/fi/aalto/cs/intellij/activities/VersionCheckerActivity.java

```diff
@@ -0,0 +1,41 @@
+package fi.aalto.cs.intellij.activities;
+
+import com.intellij.notification.Notification;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.startup.StartupActivity;
+import fi.aalto.cs.intellij.common.BuildInfo;
+import fi.aalto.cs.intellij.common.Version;
+import fi.aalto.cs.intellij.notifications.BetaVersionWarning;
+import fi.aalto.cs.intellij.notifications.Notifier;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * An activity that checks, on startup, if the version of the plugin is a pre-release, and, if that
+ * is the case, shows a {@link BetaVersionWarning}.
+ */
+public class VersionCheckerActivity implements StartupActivity {
+
+  private final Notifier notifier;
+  private final Version version;
```

Comment:
This also isn't necessarily needed as an instance variable, since it comes from the build info singleton and never changes

Reply:
>> how difficult your pseudocode test approach would be to implement

> I've understood that Mockito is specifically designed to so things like that so it shouldn't be difficult. I think @superseacat has done some research on this.

And... I was wrong :P

Apparently Mockito can't mock static methods. See e.g. https://stackoverflow.com/questions/21105403/mocking-static-methods-with-mockito